community/kernel/src/main/java/org/neo4j/kernel/internal/GraphDatabaseAPI.java

```diff
@@ -48,6 +52,43 @@
      */
     URL validateURLAccess( URL url ) throws URLAccessValidationError;
 
-    @Deprecated
     String getStoreDir();
+
+    /**
+     * Returns all relationship types currently in the underlying store.
+     * Relationship types are added to the underlying store the first time they
+     * are used in a successfully commited {@link Node#createRelationshipTo
+     * node.createRelationshipTo(...)}. Note that this method is guaranteed to
+     * return all known relationship types, but it does not guarantee that it
+     * won't return <i>more</i> than that (e.g. it can return "historic"
+     * relationship types that no longer have any relationships in the node
+     * space).
+     *
+     * @return all relationship types in the underlying store
+     */
+    ResourceIterable<RelationshipType> getAllRelationshipTypesInUseAndNot();
```

Comment:
@tinwelint we have to invent new names because methods in GGO are named the same way as in GDS. so this is the current state:
 - both GDS and GGO have methods `getAllXXX` which do different things
 - GGO additionally has methods `getAllXXXInUse` which do same as `getAllXXX` from GDS
 - both `getAllXXX` and `getAllXXXInUse` in GGO are deprecated and point to same `getAllXXX` from GDS
 - we need both `getAllXXX` from GDS and `getAllXXX` from GDO for the endpoint
 - we possibly need both to maintain public API, though it is inconsistent and deprecated; so there is a possibility to keep only GDS variants as part of public API

I'd vote to keep GDS#getAllXXX in public API but change it's behaviour to return all inUse token records. This will not impose out internal understanding about what is in use (`TokenAccess#inUse`) or not on users. For the endpoint we'll be able to filter out tokens according to `TokenAccess#inUse`. Pros: one set of simple methods that just return everything, any king of filtering could be applied later. Cons: change behaviour of public methods, though this behaviour was never properly specified and confused with methods from GGO.